module2/homework/nwdNww/nwdNww.hpp

```diff
@@ -1,11 +1,13 @@
 #pragma once
+#include <algorithm>
 
-int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+int NWD(int lhs, int rhs){
+    	while(rhs)
+		std::swap(lhs %= rhs, rhs);
+	return std::abs(lhs);
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    if (NWD(lhs, rhs)==0) {return 0;}
```

Comment:
what about 

```suggestion
   if ((lhs == 0) or (rhs == 0)) {
        return 0;
   }
```
and avoding NWW() call